### PR TITLE
make GCP project working tf

### PR DESCRIPTION
**Pull request by JamesWoolfenden**

### CODE REVIEW

File: terraform/gcp/provider.tf [hunk 1]
```diff
@@ -1,12 +1,5 @@
...
-  credentials = file(var.credentials_path)
-  project     = var.project
-  region      = var.region
-}
-
-terraform {
-  backend "gcs" {
-    credentials = var.credentials_path
-    prefix      = "terragoat/${var.environment}"
-  }
```
Comment (nimrodkor) [inline terraform/gcp/provider.tf:11]: @JamesWoolfenden Why remove the BE?

File: terraform/gcp/variables.tf [hunk 2]
```diff
@@ -1,7 +1,7 @@
-variable "credentials_path" {
-  type        = string
-  description = "Path to credentials file"
-}
+# variable "credentials_path" {
+#   type        = string
+#   description = "Path to credentials file"
+# }
```
Comment (nimrodkor) [inline terraform/gcp/variables.tf:4]:
```suggestion
```

**Review by nimrodkor — COMMENTED** [review]
Nice @JamesWoolfenden

File: terraform/gcp/provider.tf (hunk 1 above)
Comment (JamesWoolfenden) [reply terraform/gcp/provider.tf:11]: I dont think variables work in the init stage.

Comment (nimrodkor) [reply terraform/gcp/provider.tf:11]: It used to work...

**Review by nimrodkor — COMMENTED** [review]
I do believe there are some docs to update here as well, if we're changing the GCS state saving @JamesWoolfenden

File: terraform/gcp/provider.tf (hunk 1 above)
Comment (JamesWoolfenden) [reply terraform/gcp/provider.tf:11]: Theres no reference to those values, do you have env vars or a tfvars file locally? there are no vars set in this project or in the secrets?

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**